docs/tutorials/essentials/part-4-using-data.md

```diff
@@ -0,0 +1,808 @@
+
+```js title="app/store.js"
+import { configureStore } from '@reduxjs/toolkit'
+
+import postsReducer from '../features/posts/postsSlice'
+// highlight-next-line
+import usersReducer from '../features/users/usersSlice'
+
+export default configureStore({
+  reducer: {
+    posts: postsReducer,
+    // highlight-next-line
+    users: usersReducer
+  }
+})
+```
+
+### Adding Authors for Posts
+
+Every post in our app was written by one of our users, and every time we add a new post, we should keep track of which user wrote that post. In a real app, we'd have some sort of a `state.currentUser` field that keeps track of the current logged-in user, and use that information whenever they add a post.
+
+To keep things simpler for this example, we'll update our `<AddPostForm>` component so that we can select a user from a dropdown list, and we'll include that user's ID as part of the post. Once our post objects have a user ID in them, we can use that to look up the user's name and show it in each individual post in the UI.
+
+First, we need to update our `postAdded` action creator to accept a user ID as an argument, and include that in the action. (We'll also update the existing post entries in `initialState` to have a `post.user` field with one of the example user IDs.)
+
+```js title="features/posts/postsSlice.js"
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    postAdded: {
+      reducer(state, action) {
+        state.push(action.payload)
+      },
+      // highlight-next-line
+      prepare(title, content, userId) {
+        return {
+          payload: {
+            id: nanoid(),
+            title,
+            content,
+            // highlight-next-line
+            user: userId
+          }
+        }
+      }
+    }
+    // other reducers
+  }
+})
+```
+
+Now, in our `<AddPostForm>`, we can read the list of users from the store with `useSelector` and show them as a dropdown. We'll then take the ID of the selected user and pass that to the `postAdded` action creator. While we're at it, we can add a bit of validation logic to our form so that the user can only click the "Save Post" button if the title and content inputs have some actual text in them:
+
+```jsx title="features/posts/AddPostForm.js"
+import React, { useState } from 'react'
+// highlight-next-line
+import { useDispatch, useSelector } from 'react-redux'
+
+import { postAdded } from './postsSlice'
+
+export const AddPostForm = () => {
+  const [title, setTitle] = useState('')
+  const [content, setContent] = useState('')
+  const [userId, setUserId] = useState('')
+
+  const dispatch = useDispatch()
+
+  // highlight-next-line
+  const users = useSelector(state => state.users)
+
+  const onTitleChanged = e => setTitle(e.target.value)
+  const onContentChanged = e => setContent(e.target.value)
+  const onAuthorChanged = e => setUserId(e.target.value)
+
+  const onSavePostClicked = () => {
+    if (title && content) {
```

Comment:
Just a minor refactoring:
```suggestion
    if (canSave) {
```

Reply:
You're probably right, but I'm not going to make any more code changes before publishing this because it would require rebasing all the example repo commits as well as the snippet update.